mlir/lib/ExecutionEngine/ROCm/BackendUtils.cpp

```diff
@@ -79,17 +79,16 @@ void getGpuGCNArchName(hipDevice_t device, std::string &gcnArchName) {
 
 BackendUtils::BackendUtils(const std::string &defaultTriple,
                            const std::string &defaultChip,
-                           const std::string &defaultFeatures,
-                           bool systemOverride)
+                           const std::string &defaultFeatures)
     : triple(defaultTriple), chip(defaultChip), features(defaultFeatures) {
-  if (systemOverride) {
+  if (triple.empty()) {
     triple = kTargetTriple;
+  }
```

Comment:
Ok, so I can make the same check for all three are empty. This would be equivalent to previous code in mlir-rocm-runner that just set systemOverride based on same condition. This is the only place that uses BackendUtils, and now also in mlir-miopen-driver.